tests/unit/auth.cpp

```diff
@@ -407,6 +407,286 @@ TEST(AuthWithoutStorage, PermissionsMaskTest) {
+
+    ASSERT_TRUE(fga_permissions.grants().empty());
+    ASSERT_TRUE(fga_permissions.denies().empty());
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Grant("Label", LabelPermission::CREATE_DELETE);
+    fga_permissions.Deny("OtherLabel", LabelPermission::CREATE_DELETE);
+    fga_permissions.Revoke("*");
+
+    ASSERT_TRUE(fga_permissions.grants().empty());
+    ASSERT_TRUE(fga_permissions.denies().empty());
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Grant("*", LabelPermission::CREATE_DELETE);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::GRANT);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::GRANT);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::GRANT);
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Grant("*", LabelPermission::EDIT);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::GRANT);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::GRANT);
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Grant("*", LabelPermission::READ);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::GRANT);
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Deny("*", LabelPermission::READ);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::DENY);
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Deny("*", LabelPermission::EDIT);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::DENY);
+  }
+
+  {
+    FineGrainedAccessPermissions fga_permissions;
+    fga_permissions.Deny("*", LabelPermission::READ);
+
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::CREATE_DELETE), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::EDIT), PermissionLevel::DENY);
+    ASSERT_EQ(fga_permissions.Has("AnyString", LabelPermission::READ), PermissionLevel::DENY);
```

Comment:
this test is the same as the test (2 above). maybe you meant create delete instead of read